pkg/gui/controllers/custom_patch_options_menu_action.go

```diff
@@ -217,7 +217,10 @@ func (self *CustomPatchOptionsMenuAction) handlePullPatchIntoNewCommit() error {
 					_ = self.c.Helpers().Commits.PopCommitMessageContexts()
 					self.c.LogAction(self.c.Tr.Actions.MovePatchIntoNewCommit)
 					err := self.c.Git().Patch.PullPatchIntoNewCommit(self.c.Model().Commits, commitIndex, summary, description)
-					return self.c.Helpers().MergeAndRebase.CheckMergeOrRebase(err)
+					if err := self.c.Helpers().MergeAndRebase.CheckMergeOrRebase(err); err != nil {
+						return err
+					}
+					return self.c.PushContext(self.c.Contexts().LocalCommits)
```

Comment:
I think what Jesse meant [here](https://github.com/jesseduffield/lazygit/issues/3200#issuecomment-2119082513) is that if the operation results in a merge conflict, we'd still want the local commits context to be pushed. However, I think this code achieves that, because if `CheckMergeOrRebase` finds a conflict and opens the "do you want to see the conflicts or abort" popup, it returns nil.

I wasn't able to test this, because I can't come up with a scenario where extracting a patch into a new commit results in a conflict. I think this can never happen, so we're fine for two reasons. 😄

Reply:
Yes, exactly. In case of merge conflicts the `PromptForConflictHandling` function is executed and it should return `nil` as a result. When I fist picked this issue up, I was trying not to change the focus in case of error, but Jesse's comment made it everything easier 😄